desktop/ui/src/main/java/org/datacleaner/windows/AnalysisJobBuilderWindowImpl.java

```diff
@@ -751,7 +751,7 @@ protected boolean onWindowClosing() {
 
                 if (unsavedChangesChoice == 0) { // save changes
                     _saveButton.doClick();
-                } else if (unsavedChangesChoice == 2) { // cancel closing
+                } else if (unsavedChangesChoice != 1) { // cancel closing
```

Comment:
Or imagine this: what if the actual value of JOptionPane.YES_OPTION was for example 100, from the beginning of the JOptionPane existence? Then you probably wouldn't get the idea to use it in our case at all.